doc/pfmatch.md

```diff
@@ -0,0 +1,234 @@
+# Pfmatch
+
+Pfmatch is a pattern-matching language for network packets, embedded in
+Lua.  It is built on the well-known
+[pflang](https://github.com/Igalia/pflua/blob/master/doc/pflang.md)
+packet filtering language, using the fast
+[pflua](https://github.com/Igalia/pflua/blob/master/README.md) compiler
+for LuaJIT
+
+Here's an example of a simple pfmatch program that just divides up
+packets depending on whether they are UDP, TCP, or something else:
+
+```lua
+match {
+   tcp => handle_tcp
+   udp => handle_udp
+   otherwise => handle_other
+}
+```
+
+Unlike pflang filters written for such tools as `tcpdump`, a pfmatch
+program can dispatch packets to multiple handlers, potentially
+destructuring them along the way.  In contrast, a pflang filter can only
+say "yes" or "no" on a packet.
+
+Here's a more complicated example that passes all non-IP traffic, drops
+all IP traffic that is not going to or coming from certain IP addresses,
+and calls a handler on the rest of the traffic.
+
+```lua
+match {
+   not ip => forward
+   -- Drop fragmented packets.
+   ip[6:2] & 0x3fff != 0 => drop
```

Comment:
It's probably worth making it explicit that at most one clause is executed, so fragmented packets are dropped regardless of their src/dst IPs.